components/controller/src/test.rs

```diff
@@ -266,9 +266,11 @@ async fn test_delegated_incept() -> Result<(), ControllerError> {
         ));
     }
 
+    // Query with correct signature
     for qry in query {
         let signature = SelfSigningPrefix::Ed25519Sha512(km2.sign(&qry.serialize()?)?);
-        delegator.finalize_query(vec![(qry, signature)]).await?;
+        let ar = delegator.finalize_query(vec![(qry, signature)]).await?;
+        assert_eq!(ar.len(), 1);
```

Comment:
Not sure why there's 1 action required here.

Reply:
That's because delegate needs delegator's confirmation to accept own event. It asks delegator to make delegating event by sending `exn` via mailbox. Here delegator get's this request. [This comment](https://github.com/THCLab/keriox/blob/3330fbdbb60086e60d3da73ccebdceff64af8119/components/controller/src/test.rs#L251) is wrong, it asks for delegation request, not receipts.
But nice catch. It seems like the exchange is sent twice, both in `finalize_group_incept` and `finalize_exchange`. The `ar` here is the same as [here](https://github.com/THCLab/keriox/blob/3330fbdbb60086e60d3da73ccebdceff64af8119/components/controller/src/test.rs#L292). Can you remove this query and second exchange sending?